### PR TITLE
Clean up kernel/deploy to be fd-relative, drop fsync

### PR DESCRIPTION
**Pull request by cgwalters**
Work In progress, early quick review only.

### CODE REVIEW

**Comment by cgwalters** [comment]
This series passes the current non-root test suite, but I haven't run it in a real-world scenario (i.e. as root doing an upgrade) yet.

File: src/libostree/ostree-sysroot-deploy.c [hunk 1]
```diff
@@ -874,114 +875,110 @@ ostree_sysroot_write_origin_file (OstreeSysroot         *sysroot,
...
-  else
-    {
-      dir_enum = g_file_enumerate_children (bootdir, OSTREE_GIO_FAST_QUERYINFO,
-                                            G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
-                                            NULL, error);
-      if (!dir_enum)
-        goto out;
+      if (errno != ENOENT)
+        {
+          glnx_set_error_from_errno (error);
```
Comment (mbarnes) [inline src/libostree/ostree-sysroot-deploy.c]: Maybe use `glnx_set_prefix_error_from_errno()` here to match the behavior of `glnx_opendirat()`?

File: src/libostree/ostree-sysroot-deploy.c [hunk 2]
```diff
@@ -874,114 +875,110 @@ ostree_sysroot_write_origin_file (OstreeSysroot         *sysroot,
...
             }
         }
       
-      if (ret_kernel && ret_initramfs)
+      if (ret_kernel_name != NULL && ret_initramfs_name != NULL)
         break;
     }
 
-  if (ret_kernel == NULL)
+  if (!ret_kernel_name)
```
Comment (mbarnes) [inline src/libostree/ostree-sysroot-deploy.c]: Super nit-picky, but I find `ret_kernel_name == NULL` easier to read.

File: src/libostree/ostree-sysroot-deploy.c [hunk 3]
```diff
@@ -1200,50 +1199,62 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
...
+
+  dest_kernel_name = remove_checksum_from_kernel_name (tree_kernel_name, bootcsum);
+
+  if (fstatat (bootcsum_dfd, dest_kernel_name, &stbuf, 0) != 0)
     {
-      if (!gs_file_linkcopy_sync_data (tree_kernel_path, dest_kernel_path, G_FILE_COPY_OVERWRITE | G_FILE_COPY_NOFOLLOW_SYMLINKS | G_FILE_COPY_ALL_METADATA,
-                                       cancellable, error))
+      if (errno != ENOENT)
+        {
+          glnx_set_error_from_errno (error);
```
Comment (mbarnes) [inline src/libostree/ostree-sysroot-deploy.c]: Maybe use `glnx_set_prefix_error_from_errno()` here too?

File: src/libostree/ostree-sysroot-deploy.c [hunk 4]
```diff
@@ -1200,50 +1199,62 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
...
+      dest_initramfs_name = remove_checksum_from_kernel_name (tree_initramfs_name, bootcsum);
 
-      if (!g_file_query_exists (dest_initramfs_path, NULL))
+      if (fstatat (bootcsum_dfd, dest_initramfs_name, &stbuf, 0) != 0)
         {
-          if (!gs_file_linkcopy_sync_data (tree_initramfs_path, dest_initramfs_path, G_FILE_COPY_OVERWRITE | G_FILE_COPY_NOFOLLOW_SYMLINKS | G_FILE_COPY_ALL_METADATA,
-                                           cancellable, error))
+          if (errno != ENOENT)
+            {
+              glnx_set_error_from_errno (error);
```
Comment (mbarnes) [inline src/libostree/ostree-sysroot-deploy.c]: Ditto.

**Comment by mbarnes** [comment]
So far so good.  Just noted some silly nit-picky stuff, all of which is optional.

**Comment by cgwalters** [comment]
Rebased this on master, still needs some real world testing, but fixes all of the above comments.

**Comment by cgwalters** [comment]
Rebased again.

**Comment by cgwalters** [comment]
Ok, rebased and merged this.

**Pull request closed without merging**